packages/core/core-v2/src/PromiseQueue.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+
+type PromiseQueueOpts = {
+  maxConcurrent?: number
+};
+
+export default class PromiseQueue {
```

Comment:
Why do we need our own PromiseQueue, not just PQueue as you had before? (There is also one in Parcel 1, just curious the reason this time. 😉)

Reply:
Main motivation was better error handling. The primary way I am using the queue is to fill it up with some initial functions (that may add more functions as they run) and then run until the queue is empty. I wanted `queue.run()` to reject if any of the functions return a rejected promise. PQueue doesn't have `.run()`, but it does have `.pause()` and `.start()`, however `.start()` always resolves, so I decided it was time to implement my own fitting my needs :) That brings up a question I had. Should `.run()` reject in a `Promise.all()` way or a `Promise.settle()` way? Right now it behaves like `Promise.all()`.